# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: 60625635f86d
# Full Hash: 60625635f86daf1eb8d704cc4ec3d80d815c4495
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2021-01-13 21:34:39
# Description:
#   Bug 1686271 - Fix sort predicate in assertion in QuotaManager::CollectOriginsForEviction. r=dom-workers-and-storage-reviewers,janv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101562
# ==============================================================================

diff -r 5cdc1511d8ca -r 60625635f86d dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Wed Jan 13 12:09:26 2021 +0000
+++ b/dom/quota/ActorsParent.cpp	Wed Jan 13 12:34:15 2021 +0000
@@ -3932,7 +3932,7 @@
   const bool inactiveOriginsSorted =
       std::is_sorted(inactiveOrigins.cbegin(), inactiveOrigins.cend(),
                      [](const auto& lhs, const auto& rhs) {
-                       return lhs->mAccessTime <= rhs->mAccessTime;
+                       return lhs->mAccessTime < rhs->mAccessTime;
                      });
   MOZ_ASSERT(inactiveOriginsSorted);
 #endif
